src/career/api/index.ts

```diff
@@ -1,21 +1,33 @@
-import { ICareerOpportunity, IEmployment, ILocation, ISelectable, TagTypes } from 'career/models/Career';
+import { ICareerOpportunity, IEmployment, ILocation, TagTypes } from 'career/models/Career';
 import { listResource, retrieveResource } from 'common/resources';
+import { IQueryObject } from 'common/utils/queryString';
 import { ICompany } from 'core/models/Company';
 
 const API_URL = '/api/v1/career/';
 
 /** All selectable tags are wrapped in an ISelectable object */
-export type FilterJobs = [
-  ICareerOpportunity[],
-  Array<ISelectable<ICompany>>,
-  Array<ISelectable<IEmployment>>,
-  Array<ISelectable<ILocation>>
-];
-
-export const listCareerOpportunities = listResource<ICareerOpportunity>(API_URL);
-
-export const getCareerOpportunities = async () => {
-  const response = await listCareerOpportunities();
+export type FilterJobs = [ICareerOpportunity[], ICompany[], IEmployment[], ILocation[]];
+
+interface IFilters extends IQueryObject {
+  query?: string;
+  location?: string;
+  location__in?: string | string[];
+  start__gte?: string;
+  start__lte?: string;
+  end__gte?: string;
+  end__lte?: string;
+  deadline__gte?: string;
+  deadline__lte?: string;
+  company?: number;
+  company__in?: number | number[];
+  employment?: number;
+  employment__in?: number | number[];
+}
+
+export const listCareerOpportunities = listResource<ICareerOpportunity, IFilters>(API_URL);
```

Comment:
Do I understand it correctly that listResource and retrieveResource is part of a generic CRUD-set of operations?
Which is then used as part of thunk actions?

Reply:
Yes, they are an abstraction I made which maps directly to Django Rest Framework operations. It does a lot of things for you, and is supposed to make typings work very strictly for all operations, but I'm not really happy with it.

I might hope to clean it up some day, but it actually works well, so I haven't thought too much about it